samza-api/src/main/java/org/apache/samza/system/StreamSpec.java

```diff
@@ -158,8 +158,9 @@ public StreamSpec(String id, String physicalName, String systemName, int partiti
     validateLogicalIdentifier("streamId", id);
     validateLogicalIdentifier("systemName", systemName);
 
-    if (partitionCount < 1) {
```

Comment:
Please add this original validation to KafkaStreamSpec.java. It is still the proper validation for kafka.